components/scream/src/share/util/scream_common_physics_functions_impl.hpp

```diff
@@ -208,6 +208,59 @@ void PhysicsFunctions<DeviceT>::calculate_z_int(const MemberType& team,
   column_ops::template column_scan<FromTop>(team,num_levs,dz,z_int,z_surf);
 }
 
+template<typename DeviceT>
+template<typename ScalarT>
+KOKKOS_INLINE_FUNCTION
+ScalarT PhysicsFunctions<DeviceT>::calculate_vmr_from_mmr(const Real& gas_mol_weight, const ScalarT& mmr)
+{
+  using C = scream::physics::Constants<Real>;
+  constexpr Real air_mol_weight   = C::MWdry;
+
+  return mmr / (1.0 - mmr) * air_mol_weight/gas_mol_weight;
```

Comment:
Actually, I think this needs qv, and should be `mmr / (1.0 - qv) * air_mol_weight / gas_mol_weight`.